scio-bigquery/src/main/scala/com/spotify/scio/bigquery/BigQueryClient.scala

```diff
@@ -104,159 +78,82 @@ class BigQueryClient private (private val projectId: String,
 
   private val logger = LoggerFactory.getLogger(this.getClass)
 
-  private val TABLE_PREFIX = "scio_query"
-  private val TIME_FORMATTER = DateTimeFormat.forPattern("yyyyMMddHHmmss")
-  private val PERIOD_FORMATTER = new PeriodFormatterBuilder()
+  private[bigquery] val PERIOD_FORMATTER = new PeriodFormatterBuilder()
     .appendHours().appendSuffix("h")
     .appendMinutes().appendSuffix("m")
     .appendSecondsWithOptionalMillis().appendSuffix("s")
     .toFormatter
 
-  private val STAGING_DATASET_PREFIX = "scio_bigquery_staging_"
-  private val STAGING_DATASET_TABLE_EXPIRATION_MS = 86400000L
-  private val STAGING_DATASET_DESCRIPTION = "Staging dataset for temporary tables"
-
-  private val DEFAULT_LOCATION = "US"
+  private[scio] def isCacheEnabled: Boolean = BigQueryClient.isCacheEnabled
 
-  private def isInteractive =
-    BigQueryClient.priority
-      .map(_ == "INTERACTIVE")
-      .getOrElse(
-        Thread
-          .currentThread()
-          .getStackTrace
-          .exists { e =>
-            e.getClassName.startsWith("scala.tools.nsc.interpreter.") ||
-            e.getClassName.startsWith("org.scalatest.tools.")
-          }
-      )
+  /** Services */
+  private[scio] val tables = new TableService(projectId, bigquery, credentials)
```

Comment:
This will cause lazy `credentials` to evaluate thus to create GCP credentials, and this will not work in untrusted PR builds. Also you probably don't want this to be eagerly evaluated anyway.